third_party/move/move-compiler-v2/src/pipeline/ability_checker.rs

```diff
@@ -281,6 +300,14 @@ fn check_bytecode(target: &FunctionTarget, code_offset: CodeOffset, bytecode: &B
                 ),
                 ReadRef => check_read_ref(target, srcs[0], &loc),
                 WriteRef => check_write_ref(target, code_offset, srcs[0], &loc),
+                Eq => {
```

Comment:
Why do we think this is necessary and equal would behave different?

In any case if it does behave different then this is because we do not 'move' arguments into those operations which does not need to be special cased here.

Reply:
This is necessary because `==` does consume the arguments (mentioned in the v1 compiler warning), and destroys it inside. So the signature of it should be like `native fun eq<T: drop>(x: T, y: T): bool`.
This is also what the bytecode verifier expected.
https://github.com/aptos-labs/aptos-core/blob/ea133dc57bf2f51e3e9dfa237b1a9d4ee4ddf77d/third_party/move/move-bytecode-verifier/src/type_safety.rs#L729

Reply:
Well then we are all good because == doesn't behave different than any other function. At least in the new copy/move we treat them all the same (Function/whatever op). The fix before the new copy/move should also not be here by special casing this, rather its regular behavior of functions.